registry/registry.go

```diff
@@ -556,19 +573,37 @@ func createVersion(c *Space, db *kivik.DB, ver *Version, attachments []*kivik.At
 	versionChannel := GetVersionChannel(ver.Version)
 	for _, channel := range []Channel{Stable, Beta, Dev} {
 		if channel >= versionChannel {
-			key := cache.Key(c.prefix + "/" + ver.Slug + "/" + channelToStr(channel))
+			key := cache.Key(c.Prefix + "/" + ver.Slug + "/" + channelToStr(channel))
 			cacheVersionsLatest := viper.Get("cacheVersionsLatest").(cache.Cache)
 			cacheVersionsList := viper.Get("cacheVersionsList").(cache.Cache)
 			cacheVersionsLatest.Remove(key)
 			cacheVersionsList.Remove(key)
 		}
 	}
 
+	// Storing the attachments to swift (screenshots, icon, partnership_icon)
+	basePath := filepath.Join(ver.Slug, ver.Version)
+
+	prefix := c.Prefix
+	if prefix == "" {
+		prefix = consts.DefaultSpacePrefix
+	}
+
 	for _, att := range attachments {
-		ver.Rev, err = db.PutAttachment(ctx, ver.ID, ver.Rev, att)
+		f, err := sc.ObjectCreate(prefix, filepath.Join(basePath, att.Filename), false, "", att.ContentType, nil)
+		if err != nil {
+			return err
+		}
+		content, err := ioutil.ReadAll(att.Content)
 		if err != nil {
 			return err
 		}
+		_, err = f.Write(content)
+		if err != nil {
+			return err
+		}
+
+		f.Close()
```

Comment:
You should put `defer f.Close()` sooner (L597) to avoid keeping `f` opened if there is an error when reading the content.